src/app/campaign/store/effects/campaign.effects.ts

```diff
@@ -151,11 +152,47 @@ export class CampaignEffects {
     catchError(error => of(new campaignActions.CampaignDupByIdFailure({ error })))
   );
 
+  @Effect({ dispatch: false })
+  duplicateCampaign$ = this.actions$.pipe(
+    ofType(ActionTypes.CAMPAIGN_DUPLICATE),
+    tap((action: campaignActions.CampaignDuplicate) => {
+      const { campaign, flights } = action.payload;
+      const state = {
+        campaign,
+        flights: flights.map(flight => ({
+          ...flight,
+          startAt: undefined,
+          endAt: undefined
+        }))
+      };
+      this.router.navigate(['/campaign', 'new'], { state });
+    })
+  );
+
```

Comment:
alternatively, I feel like the one effect for action CAMPAIGN_DUP_FROM_FORM could have done the router.navigate as well as the toastr.success. It's a weird one since it's using effects but actually synchronous since we don't wait on the navigation or anything and just expect that the app reactively works when these things happen, so I'm really just mentioning this as a thought.